### ISSUE TITLE
Suggestion - Mechanic Role Updates

### ISSUE THREAD

**Issue by CaptSpider4** [InZidiuZ/op-framework-issue-tracker#2593]
### Assurance

I have read the `README.md`

### Summary

Well as you know Every role has 3 parts Job, Department, Position. My suggestion is to add more departments based on shops names. The base concept of this suggestion is to restrict mechanics from accessing different mechanic shops which they are not part of in other words Mosley's Mechanic shouldn't be able to do anything in Hayes.

Job - Mechanic will have access to following things
* Mechanic Helpline
* Impound Lots and tow system 

Positions - Will be for salary

Department Names will be based on that shop for example Mosley, Hayes etc. Or repair shop number if names is not possible.

Department will also have the current one **Los Santos Customs** with which they can access all repair shops for the subdivisions who don't want to restrict people to use particular shop and for mechanics who work for all shops part time (Freelance mechanic)

Departments - Will have access to following things
* Locks for that shop
* Review vehicles in that shop
* Use workbench of that shop
* Buy materials in that shop
* Access locker and cloth change spot of that shop 

### Reason

Basically, this suggestion is to stop people from Fail RP. It's hard to keep an eye on mechanics who repair vehicles in different repair shop. Even there is no direct way to check in which repair shop they used to repair vehicles from Server Logs. In Indian subdivision we currently have 80 to 100 mechanics from 4 shops. Monitoring people about this especially during non-prime time is extremely hard.

**Comment by superiorchicken007** [comment]
> Basically, this suggestion is to stop people from Fail RP. 
IMO this can all be handled under each subdivisions on the rules for Fail RP.

I do not do coding but this seems like it might be too much for each shop to have their own stuff locked to a department as much as I agree it would be nice but unnecessary for the amount of work it could be.   Plus If/When the department would change people would loose everything in their lockers as it would be with that department if they would implement the different business's departments.

**Comment by CloudTheWolf** [comment]
This is probably something that wont get done as, at least in the international server, state will sometimes ask one shop to help run another shop when the first open / change ownership, or one shop may ask to another to help staff the shop whilst they fill positions.

**Comment by coalaura** [comment]
Closing due to reasons stated above
